src/ContactsAssistant/menu.py

```diff
@@ -1,44 +1,151 @@
 """Class Menu module"""
 
 import difflib
+import argparse
 from enum import Enum
 from collections import namedtuple
 from constants import MENU_BORDER
 from utils import format_cmd, format_param
 
-Item = namedtuple('Item', ['min_required_params', 'param_list', 'hint'])
+Command = namedtuple("Command", ["min_required_params", "param_list", "hint"])
+Parametr = namedtuple(
+    "Parametr",
+    ["name", "required", "hint", "choices"],
+    defaults=(None, False, None, None),
+)
+
 
 class Menu(Enum):
-    '''Class'''
-    HELLO                   = Item(0, [], 'to show command list')
-    ADD_CONTACT             = Item(2, ['CONTACT_NAME', 'PHONE', 'EMAIL'], 'to add a new contact')
-    UPDATE_CONTACT          = Item(3, ['CONTACT_NAME','OLD PHONE','NEW PHONE'], 'to update a phone')
-    DELETE_CONTACT          = Item(1, ['CONTACT_NAME'], 'to delete contact')
-    SET_CONTACT_BIRTHDAY    = Item(2, ['CONTACT_NAME','BIRTHDAY DATE'], 'to set birthday')
-    GET_CONTACT_BIRTHDAY    = Item(1, ['CONTACT_NAME'], 'to show birthday')
-    GET_CONTACT_BY_NAME     = Item(1, ['CONTACT_NAME'], 'to find a contact by name')
-    GET_CONTACT_BY_PHONE    = Item(1, ["PHONE"], 'text": "to find a contact by phone')
-    GET_CONTACT_BY_EMAIL    = Item(1, ["EMAIL"], 'text": "to find a contact by email')
-    GET_ALL_CONTACTS        = Item(0, [], 'to view a full contact list')
-    GET_UPCOMING_BIRTHDAYS  = Item(0, [], 'to show all birthdays in this week')
-    UPDATE_CONTACT_EMAIL    = Item(2, ['CONTACT_NAME', 'EMAIL'], 'to update email')
-    NOTE_ADD                = Item(2, ['NOTE_SUBJECT','NOTE_TEXT'], 'to add or update note')
-    NOTE_DEL                = Item(1, ['NOTE_SUBJECT'], 'to delete note')
-    NOTE_TAG                = Item(2, ['NOTE_SUBJECT','TAG'], 'to add tag to note')
-    NOTE_TAG_DEL            = Item(2, ['NOTE_SUBJECT','TAG'], 'to delete note tag')
-    NOTE_ALL                = Item(0, [], 'to view a full note list')
-    EXIT                    = Item(0, [], 'to app close')
-    CLOSE                   = Item(0, [], 'to app close')
+    """Class"""
+
+    HELLO = Command(0, [], "to show command list")
+    ADD_CONTACT = Command(
+        2,
+        [
+            Parametr("name", True, "contact name"),
+            Parametr("phone", False, "contact phone number in format XXXXXXXXXX"),
```

Comment:
I think that it would be better to do phone as a required parametr

Reply:
set as required